app/src/common/shared/com/igalia/wolvic/utils/AnimationHelper.java

```diff
@@ -17,6 +17,13 @@ public class AnimationHelper {
     public static final long FADE_ANIMATION_DURATION = 150;
     
     public static void fadeIn(View aView, long delay, final Runnable aCallback) {
+        if (aView.getVisibility() == View.VISIBLE && aView.getAlpha() == 1) {
+            if (aCallback != null) {
+                aCallback.run();
+            }
+            return;
+        }
```

Comment:
Mind explaining what the changes in this file are for?

Reply:
When working on this PR I ran into a few cases where we tried to animate a widget that was already being shown or hidden. This caused an ugly effect in the UI.